sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -192,7 +193,10 @@ private[javasdk] final class ActionsImpl(_system: ActorSystem, services: Map[Str
               Future.successful(handleUnexpectedException(service, in, ex))
           }
         fut.andThen { case _ =>
-          span.foreach(_.end())
+          span.foreach { s =>
+            mdc.remove(Telemetry.TRACE_ID)
```

Comment:
This is removing from a copy of the MDC, so doesn't affect the actual MDC.

It's also removing in a future callback, which may not be the same thread as where the trace id was added to the MDC. This means that the trace id can be 'leaked' — other logs when something runs on that same original thread will be incorrectly logged with the trace id.

Reply:
done